airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -176,6 +188,62 @@ def _doc_upload(
     return doc_uploaded, doc_blob_id
 
 
+def _file_upload(
+    local_path: Path,
+    gcp_connector_dir: str,
+    bucket: storage.bucket.Bucket,
+    *,
+    upload_as_version: str | Literal[False],
+    upload_as_latest: bool,
+    skip_if_not_exists: bool = True,
+) -> tuple[tuple[bool, str], tuple[bool, str]]:
```

Comment:
💅 Might be high time to formalize that tuple
```suggestion
) -> tuple[MaybeUpload, MaybeUpload]:
```